### PR TITLE
adding user models

### PR DESCRIPTION
**Pull request by Swetlana-K**
adding user models

### CODE REVIEW

File: apps/user/models.py [hunk 1]
```diff
@@ -1,8 +1,50 @@
...
+    birthday = models.DateField()
+    GENDER_CHOICES = [
+        ('M', 'Male'),
+        ('F', 'Female'),
+        ('O', 'Other'),
+    ]
+    gender = models.CharField(max_length=1, choices=GENDER_CHOICES)
+    phone = models.CharField(max_length=20)
+    profile_email = models.EmailField()
+
```
Comment (Termuellinator) [inline apps/user/models.py:26]: I think we should include the language here as a many to many field:
`languages = models.ManyToManyField("user.Language", through = "UserLanguage")` should do the trick i think?

File: apps/user/models.py [hunk 2]
```diff
@@ -1,8 +1,50 @@
...
+    title = models.CharField(max_length=100, null=True,
+                             blank=True, help_text="Your profession")
     bio = models.TextField(default='', blank=True)
+
+    mfa_secret = models.CharField(max_length=255, default='')
+    date_joined =  models.DateField(default=timezone.now)
+
+
+class UserProfile(models.Model):
+    id = models.AutoField(primary_key=True)
```
Comment (Termuellinator) [inline apps/user/models.py]: we don't need to specify the id as a explicit field

File: apps/user/models.py [hunk 3]
```diff
@@ -1,8 +1,50 @@
...
+        ('F', 'Female'),
+        ('O', 'Other'),
+    ]
+    gender = models.CharField(max_length=1, choices=GENDER_CHOICES)
+    phone = models.CharField(max_length=20)
+    profile_email = models.EmailField()
+
+
+class Language(models.Model):
+    id = models.AutoField(primary_key=True)
```
Comment (Termuellinator) [inline apps/user/models.py]: we don't need to specify the id as a explicit field

File: apps/user/models.py [hunk 4]
```diff
@@ -1,8 +1,50 @@
...
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    name = models.CharField(max_length=255)
+    hometown = models.CharField(max_length=255)
+    slogan = models.CharField(max_length=255)
+    birthday = models.DateField()
+    GENDER_CHOICES = [
+        ('M', 'Male'),
+        ('F', 'Female'),
+        ('O', 'Other'),
```
Comment (Termuellinator) [inline apps/user/models.py]: Maybe use "diverse" instead of "other"?

**Review by Termuellinator — CHANGES_REQUESTED** [review]
Other than the comments, LGTM :)

File: apps/user/models.py [hunk 5]
```diff
@@ -1,8 +1,50 @@
...
-# Create your models here.
 
 class User(AbstractUser):
-    title = models.CharField(max_length=100, null=True, blank=True, help_text="Your profession")
+    title = models.CharField(max_length=100, null=True,
+                             blank=True, help_text="Your profession")
     bio = models.TextField(default='', blank=True)
+
+    mfa_secret = models.CharField(max_length=255, default='')
+    date_joined =  models.DateField(default=timezone.now)
```
Comment (Termuellinator) [inline apps/user/models.py]: We removed those two fields in a previous commit

File: apps/user/models.py [hunk 6]
```diff
@@ -1,8 +1,50 @@
...
+    def __str__(self):
+        return self.lang
+
+class UserLanguage(models.Model):
+    userprofile = models.ForeignKey(UserProfile, on_delete=models.CASCADE)
+    language = models.ForeignKey(Language, on_delete=models.CASCADE)
+    level_choices = [
+        ('Beginner', 'Beginner'),
+        ('Intermediate', 'Intermediate'),
+        ('Advanced', 'Advanced'),
```
Comment (Termuellinator) [inline apps/user/models.py]: I think we had `learning`, `fluent` and `preferred` for this in our discussion.

**Review by Termuellinator — CHANGES_REQUESTED** [review]
Some more things that i noticed

File: apps/user/models.py [hunk 7]
```diff
@@ -1,8 +1,50 @@
...
+class Language(models.Model):
+    id = models.AutoField(primary_key=True)
+    lang = models.CharField(max_length=255)
+    
+    def __str__(self):
+        return self.lang
+
+class UserLanguage(models.Model):
+    userprofile = models.ForeignKey(UserProfile, on_delete=models.CASCADE)
+    language = models.ForeignKey(Language, on_delete=models.CASCADE)
```
Comment (Termuellinator) [inline apps/user/models.py:37]:
```
class Meta:
    constraints = [
        models.UniqueConstraint(fields=['userprofile', 'language'], name='unique language')
    ]
```

File: apps/user/models.py [hunk 8]
```diff
@@ -1,8 +1,53 @@
 from django.db import models
 from django.contrib.auth.models import AbstractUser
+from django.utils import timezone
```
Comment (Termuellinator) [inline apps/user/models.py]: I'm just noticing: i don't think you are actually using timezone anywhere, or am i missing something?

**Review by Termuellinator — APPROVED** [review]
LGTM besides the timezone import

File: apps/user/models.py (hunk 8 above)
Comment (Swetlana-K) [reply apps/user/models.py]: But you don't miss a thing ;).
I have overlooked it. It comes from the old code:
date_joined = models.DateField(default=timezone.now)

**Pull request merged**